crates/typst/src/math/stretch.rs

```diff
@@ -73,9 +73,15 @@ pub(super) fn stretch_fragment(
         _ => return,
     };
 
-    let Some(axis) = axis.or_else(|| stretch_axis(ctx, &glyph)) else {
+    // Return if we attempt to stretch along an axis which isn't stretchable,
+    // so that the original fragment isn't modified.
+    let Some(stretch_axis) = stretch_axis(ctx, &glyph) else {
         return;
     };
+    if axis.is_some_and(|x| x != stretch_axis) {
+        return;
+    }
+    let axis = axis.unwrap_or(stretch_axis);
```

Comment:
There's probably a nicer way to structure this bit that I can't think of right now.

Reply:
`axis != Some(stretch_axis)` ? Maybe one big match statement

Reply:
`axis != Some(stretch_axis)` ? Maybe one big match statement

Reply:
It shouldn't return if axis is `None` though. A match might be nicer.

Reply:
yeah that works.